packages/flutter/test/material/search_anchor_test.dart

```diff
@@ -3016,6 +3016,67 @@ void main() {
     expect(opacityWidget.opacity, 0.38);
   });
 
+  testWidgets('Check SearchAnchor opacity when disabled', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      home: Center(
+        child: Material(
+          child: SearchAnchor(
+            isFullScreen: true,
+            enabled: false,
+            builder: (BuildContext context, SearchController controller) {
+              return const Icon(Icons.search);
+            },
+            headerHeight: 32,
+            suggestionsBuilder: (BuildContext context, SearchController controller) {
+              return <Widget>[];
+            },
+          ),
+        ),
+      ),
+    ));
+
+    final Finder searchBarFinder = find.byType(SearchAnchor);
+    expect(searchBarFinder, findsOneWidget);
+    final Finder opacityFinder = find.descendant(
+      of: searchBarFinder,
+      matching: find.byType(AnimatedOpacity),
+    );
+    expect(opacityFinder, findsOneWidget);
+    final AnimatedOpacity opacityWidget = tester.widget<AnimatedOpacity>(opacityFinder);
+    expect(opacityWidget.opacity, 0.38);
+  });
+
+  testWidgets('SearchAnchor tap failed when disabled', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      home: Center(
+        child: Material(
+          child: SearchAnchor(
+            isFullScreen: true,
+            enabled: false,
+            builder: (BuildContext context, SearchController controller) {
+              return const Icon(Icons.search);
+            },
+            headerHeight: 32,
+            suggestionsBuilder: (BuildContext context, SearchController controller) {
+              return <Widget>[];
+            },
+          ),
+        ),
+      ),
+    ));
+
+    final Finder searchBarFinder = find.byType(SearchAnchor);
+    expect(searchBarFinder, findsOneWidget);
+    bool tapFailed = false;
+
+    try {
+      await tester.tap(searchBarFinder);
+    } catch (_) {
+      tapFailed = true;
+    }
+    expect(tapFailed, isTrue);
```

Comment:
Instead of try/catch, could you set warnIfMissed to false? See https://main-api.flutter.dev/flutter/flutter_test/WidgetController/tap.html.

Reply:
the point here is to get the error and not to avoid it, and that is why i used it instead of warnIfMissed

Reply:
If i use warnIfMissed = false, i cant be sure the tap failed

Reply:
There's some concern that the tap could hit the button yet cause some other sort of error, and this test would still pass. Could you use warnIfMissed and then expect that the suggestions aren't visible or the suggestionsBuilder hasn't been called?

Reply:
Im really not sure on how to obtain a clear test with warnifmissed as this could mean anything...
Also if the error won't happen, how can i check that the tap was "blocked"